app/javascript/mastodon/features/compose/containers/compose_form_container.js

```diff
@@ -18,6 +18,24 @@ import ComposeForm from '../components/compose_form';
 
 const urlLikeRegex = /^https?:\/\/[^\s]+\/[^\s]+$/i;
 
+const processPasteOrDrop = (transfer, e, dispatch) => {
+  if (transfer && transfer.files.length === 1) {
+    dispatch(uploadCompose(transfer.files));
+    e.preventDefault();
+  } else if (transfer && transfer.files.length === 0) {
```

Comment:
Having an early return `if (!transfer)` might make the conditionals a bit more readable, but the original code was written the same way, so no need to change.